lib/OpenQA/WebAPI/Controller/Step.pm

```diff
@@ -455,7 +455,9 @@ sub save_needle_ajax {
         return $self->render(json => {error => "Error creating/updating needle: $error"});
     }
 
-    my $job          = $self->app->schema->resultset("Jobs")->find($self->param('testid'));
+    my $job = $self->app->schema->resultset('Jobs')->find($self->param('testid'))
+      or return $self->render(json => {error => 'Specified job does not exist.'}, status => 404);
```

Comment:
please use the form we use everywhere for this code pattern and split this into 2 statements.